src/gboml/scope.py

```diff
@@ -95,6 +115,25 @@ def __getitem__(self, item):
         return out
 
 
+@dataclass
+class GeneratedRValueScope(Scope, Generic[U]):
+    parent: "Scope" = field(repr=False)
+    ast: U
+    name: str = field(init=False, default=None)
+    path: list[str] = field(init=False)
+    content: dict[str, "Scope"] = field(init=False)
+
+    def __post_init__(self):
+        self.ast.scope = self
+        self.path = self.parent.path
+        self.content = self.parent.content
+
+    # needed post_post_init because we need parent's scope fully filled in to update it with keys and check if intersects
+    def _finalize_init(self):
+        self.content = self.parent.content.copy()
```

Comment:
We should avoid copying if we can

Reply:
Right.
I'll define some logic in `__getitem__()` to handle that (thus `_add_unnamed_to_scope()` function will become irrelevant and will be deleted).
But I think I still need the `_finalize_init()` to check if the value is already in scope.